### DISCUSSION TITLE
java.lang.constant.Constable should be treated as a transparent trait to help Scala.js

### DISCUSSION THREAD

**Discussion by sjrd** [scala/scala3#17853] [Feature Requests]
See the discussion at https://github.com/typelevel/shapeless-3/pull/1#issuecomment-847197636

On JDK 12+, `java.lang.constant.Constable` can come up as an inferred LUB of primitive classes like `String` and `Integer`. This is problematic on Scala.js (and in the future in Scala Native) since the API of `Constable` cannot be implemented, causing linking errors.

In Scala 2, there is not much we can do about it except using explicit type ascriptions to avoid it being inferred. In Scala 3 however, this would easily be solved by treating `Constable` as a transparent trait, like is done (IIRC) for `java.lang.Comparable`.

**Comment by joroKr21** [comment]
Note that the problem in shapeless is caused by the compiler generated `productElement` of
```scala
case class ValueTypeable[T, B](cB: Class[B], describe: String)
```
